Translate 04-advanced-sveltekit/02-page-options

suhyeoonn opened the pull request:
Advanced sveltekit의 page options 부분 번역하였습니다.
Close https://github.com/Svelte-Korea/learn.svelte.kr/issues/26

suhyeoonn committed:
Translate 04-advanced-sveltekit/02-page-options

D0ri123 commented on content/tutorial/04-advanced-sveltekit/02-page-options/01-page-options/README.md:
```diff
@@ -1,16 +1,16 @@
+- `trailingSlash` — URL의 트레일링 슬래시를 제거, 추가 또는 무시할지 여부 
 
-In the following exercises, we'll learn about each of these in turn.
+다음 연습에서는 이들 각각에 대해 차례로 알아봅니다.
```
뭔가 문장에서 시제가 부자연스러운 것 같아서요! 
'알아봅니다.' 보다는 '**알아보겠습니다.**' 또는 '**알아볼것입니다.**' 라는 표현은 어떨까요?

D0ri123 commented on content/tutorial/04-advanced-sveltekit/02-page-options/02-ssr/README.md:
```diff
@@ -1,14 +1,14 @@
 ---
 
-Server-side rendering (SSR) is the process of generating HTML on the server, and is what SvelteKit does by default. It's important for performance and [resilience](https://kryogenix.org/code/browser/everyonehasjs.html), and is very beneficial for search engine optimization (SEO) — while some search engines can index content that is rendered in the browser with JavaScript, it happens less frequent ...
+서버 사이드 렌더링(SSR)은 서버에서 HTML을 생성하는 프로세스로, 스벨트킷이 기본적으로 수행하는 작업입니다. 이는 성능과 [복원력](https://kryogenix.org/code/browser/everyonehasjs.html)에 중요하며 검색 엔진 최적화(SEO)에서도 중요한 역할을 합니다. 일부 검색 엔진은 자바스크립트로 렌더링 된 콘텐츠를 색인화할 수 있지만 빈도와 안정성이 낮습니다.
```
'일부 검색 엔진~' 이라는 문장 앞에 `반면에`라는 접속사를 붙이면 더 좋을 거 같다는 생각이 들었습니다! 
문장을 읽어봤을 때 잘 모르는 사람은 스벨트킷에 속한 일부 검색 엔진이라고 오해할 수도 있을 것 같아요~!

suhyeoonn replied:
혹시 아래처럼 수정하는 건 어떠세요??
`일부 검색 엔진은 자바스크립트로 렌더링 된 콘텐츠를 색인화할 수 있지만 빈도와 안정성이 낮습니다.`
-> 자바스크립트로 렌더링 된 콘텐츠를 색인화할 수 있는 검색 엔진도 있지만 빈도와 안정성이 낮습니다.

D0ri123 replied:
오 넵!! 더 자연스럽고 좋은 것 같아요!

D0ri123 commented on content/tutorial/04-advanced-sveltekit/02-page-options/02-ssr/README.md:
```diff
@@ -1,14 +1,14 @@
+서버 사이드 렌더링(SSR)은 서버에서 HTML을 생성하는 프로세스로, 스벨트킷이 기본적으로 수행하는 작업입니다. 이는 성능과 [복원력](https://kryogenix.org/code/browser/everyonehasjs.html)에 중요하며 검색 엔진 최적화(SEO)에서도 중요한 역할을 합니다. 일부 검색 엔진은 자바스크립트로 렌더링 된 콘텐츠를 색인화할 수 있지만 빈도와 안정성이 낮습니다.
 
-That said, some components _can't_ be rendered on the server, perhaps because they expect to be able to access browser globals like `window` immediately. If you can, you should change those components so that they _can_ render on the server, but if you can't then you can disable SSR:
+그럼에도 불구하고, 일부 컴포넌트는 서버에서 렌더링 되기 어려울 수 있습니다. 이는 아마도 해당 컴포넌트가 즉시 `window`와 같은 브라우저 전역에 액세스할 수 있을 것으로 기대하기 때문입니다. 가능하다면, 이러한 컴포넌트를 서버에서 렌더링될 수 있도록 변경하는 것이 좋습니다. 그러나 그게 불가능한 경우에는 SSR을 비활성화할 수 있습니다.
```
'그럼에도 불구하고'라는 말은 너무 직역한 거 같아서 문장 읽기에 어색하게 느껴질 수도 있을 거 같아서요~!! 
혹시 `그렇다 해도`는 어떨까욥?

D0ri123 commented on content/tutorial/04-advanced-sveltekit/02-page-options/02-ssr/README.md:
````diff
@@ -1,14 +1,14 @@
 export const ssr = false;
 ```
 
-> Setting `ssr` to `false` inside your root `+layout.server.js` effectively turns your entire app into a single-page app (SPA).
````
root가 최상단 위치를 뜻하는 거라고 생각했는데요! 혹시 이게 맞다면 `루트 +layout.server.js` 보다는 `최상단의 +layout.server.js`는 어떨까요?

suhyeoonn replied:
코멘트 감사합니다! 그런데 최상단이라는 의미는 맞지만 `루트 디렉토리`처럼 루트가 저는 더 익숙한 표현 같아서요..!
혹시 `root`를 `최상단`으로 번역하는 것이 규칙이라면 수정하겠습니다~!! 그런데 만약 아직 정해진 부분은 아니라면 루트라고 하는 건 어떤지 검토 부탁드려도 될까요..?
Next.js에서도 동일한 개념이 있어서 어떻게 번역 되었는지 찾아봤는데, 여기서도 `루트`로 되어있어서 공유드려요!
https://github.com/Nextjs-kr/Nextjs.kr/blob/main/docs/02-app/01-building-your-application/01-routing/02-pages-and-layouts.mdx#L108

D0ri123 replied:
아하!! 아직 규칙은 없는 걸로 알고 있습니다! 그러면 `루트에 있는 +layout.svelte.js`는 어떨까요?
중간에 수식 없이 `루트` + `+layout.server.js` 명사들끼리 붙어있는 부분이 어색해보여서 수정 제안드렸었습니다~!
루트를 넣어도 저는 좋을 것 같습니다!

suhyeoonn replied:
넵~! 그럼 `루트에 있는 +layout.svelte.js`로 수정할게요! 감사합니다☺️

D0ri123 commented on content/tutorial/04-advanced-sveltekit/02-page-options/04-prerender/README.md:
```diff
@@ -1,22 +1,22 @@
+사전 렌더링이란 각 요청에 대해 동적으로 생성하는 것이 아니라 빌드 시 한 번만 페이지에 대한 HTML을 생성하는 것을 의미합니다.
 
-The advantage is that serving static data is extremely cheap and performant, allowing you to easily serve large numbers of users without worrying about cache-control headers (which are easy to get wrong).
+이것의 장점은 정적 데이터를 제공하는 것이 매우 저렴하고 성능이 우수하므로 캐시 제어 헤더를 걱정할 필요 없이 대량의 사용자에게 쉽게 서비스를 제공할 수 있다는 것입니다. (캐시 제어 헤더를 올바르게 설정하기 어려운 부분이 있습니다.)
```
캐시 제어 해더가 설정하기 어렵지만 해당 기능을 통해서 캐시 제어 헤더를 걱정하지 않아도 된다는 점이 좋다는 걸로 이해했는데요!
괄호 안에 '캐시 제어 헤더를'이 아닌 '캐시 제어 헤더는'이라고 수정하면 의미가 더 가까워질 거 같습니다 :)

suhyeoonn committed:
피드백 반영

01 page-options `알아봅니다` -> `알아보겠습니다` 수정
02 ssr `그럼에도 불구하고` -> `그렇다 해도` 수정
04 prerender `헤더를` -> `헤더는` 수정

suhyeoonn committed:
ssr 피드백 반영
